### ISSUE TITLE
Use gulp CLI to get tasks

### ISSUE THREAD

**Issue by flying-sheep** [lkytal/quickTask#8]
gulp 4 supports multiple styles of defining task, and even if you include parsing those, regexes just aren’t up to the task. what if someone did e.g. `import tasks from 'gulp'; tasks.task(...)`?

just depend on `gulp-cli` and use the output of `gulp --tasks` or `gulp --tasks-json`. (Sadly there’s no JS API to get the tasks. `gulp-cli` is structured in a way that you can only run it, not access it programmatically)

**Comment by lkytal** [comment]
Thanks for your suggestion. I've complished code with gulp-cli, however, not as stable as expect, so I may delay this feature until gulp 4 released.

**Comment by flying-sheep** [comment]
what do you mean by not stable? isn’t it just:

```js
const tasks = child_process.execFileSync('gulp', ['--tasks']).split('\n')
```

**Comment by lkytal** [comment]
Exactly, however, It sometimes takes an unexpected long time to parse (5s - 10s) during tests, still unsure why. I'm considering using gulp-cli at first and fallback to regex method when timeout.

**Comment by flying-sheep** [comment]
too bad. is there an easy way to hack in profiling to tell us what the performance problem is?

**Comment by lkytal** [comment]
Well, I will give it a try.

**Comment by flying-sheep** [comment]
the gulp guys also said in gulpjs/gulp-cli#112 that apparently gulp itself can be called from JS to list tags.

my reasoning why i proposed to use gulp-cli is stated in that issue (and not answered at the time of writing)

**Comment by flying-sheep** [comment]
ah, so you fixed it in f07a80b9b597b9f777b03a8c6456fabc40aa1ec2

thank you!
